api/proto/teleport/legacy/types/types.proto

```diff
@@ -3001,6 +3006,16 @@ message KubeAWS {
   string Name = 3 [(gogoproto.jsontag) = "name,omitempty"];
 }
 
+// KubeGCP contains the GCP information about the cluster.
+message KubeGCP {
+  // Location is a GKE cluster location.
+  string Location = 1 [(gogoproto.jsontag) = "location,omitempty"];
```

Comment:
What's "location"? Maybe call it "region" for consistency with other providers?

Reply:
Location is the GCP name for the region or zone.
In GCP you can either place a cluster in a region  - across several availability zones, or in a single zone (AZ). 
To aggregate GCP mentions them as Location.
Should I rename it to region?

Reply:
So the user can specify either a region or a zone here? Could you give a couple of examples?

Reply:
Yes.

A cluster located in the `europe-west1` region is a multi-zone cluster (AWS calls them High Availability clusters)
Can have nodes in:
- `europe-west1-b`
- `europe-west1-c`
- `europe-west1-d`
- ...

A cluster located in the `europe-west1-b` zone only has nodes in `europe-west1-b`.

While creating a GKE cluster you are asked if you prefer a Zonal or a Regional cluster. Depending on the option you select you can choose between a list of GCP regions or a list of all GCP zones that support GKE.

If you filter with the `europe-west1` location, it will list every Regional cluster within that region but it won't include any zonal cluster you might have in the same region.  To include those, you need to specify the zone you want to include and the GCP List will exclude the regional clusters.

In order to include both you need to specify `Locations: ["europe-west1","europe-west1-b"]` in the watcher